File: examples/eApi v1.9/java/eAPI-v19-connector/src/main/java/cz/monetplus/mips/eapi/v19/connector/entity/MallpayOrderItemRef.java
package cz.monetplus.mips.eapi.v19.connector.entity;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Data
@EqualsAndHashCode(callSuper = false)
@JsonInclude(value = JsonInclude.Include.NON_NULL)
@AllArgsConstructor @NoArgsConstructor
public class MallpayOrderItemRef extends ApiBase  implements Signable {
    private String code;
    private String ean;
    private String name;
    private String type;
    private Long quantity;

    @Override
    public String toSign() {
        StringBuilder sb = new StringBuilder();
        add(sb, code);
        add(sb, ean);
        add(sb, name);
        add(sb, type);
        add(sb, quantity);
        deleteLast(sb);
        return sb.toString();
    }
}

File: examples/eApi v1.9/dotNET/CsobGTWDemo/CsobGatewayClientExample/Communication/DataObjects/MallpayOrderItemRef.cs
using System.Text;
using Newtonsoft.Json;

namespace CsobGatewayClientExample.Communication.DataObjects;

public class MallpayOrderItemRef : ApiBase
{
    [JsonProperty("code")] public string? Code { get; set; }

    [JsonProperty("ean")] public string? Ean { get; set; }

    [JsonProperty("name")] public string? Name { get; set; }

    [JsonProperty("type")] public string? Type { get; set; }

    [JsonProperty("quantity")] public long Quantity { get; set; }


    public string? ToSign()
    {
        var sb = new StringBuilder();
        Add(sb, Code);
        Add(sb, Ean);
        Add(sb, Name);
        Add(sb, Type);
        Add(sb, Quantity);
        DeleteLast(sb);
        return sb.ToString();
    }
}